testing/camunda-process-test-java/src/main/java/io/camunda/process/test/impl/extension/CamundaProcessTestContextImpl.java

```diff
@@ -258,33 +287,50 @@ public void completeUserTask(final UserTaskSelector userTaskSelector) {
   public void completeUserTask(
       final UserTaskSelector userTaskSelector, final Map<String, Object> variables) {
     final CamundaClient client = createClient();
-    final AtomicReference<Long> userTaskKey = new AtomicReference<>();
+    final UserTask userTask = awaitUserTask(userTaskSelector, client);
 
-    awaitBehavior.untilAsserted(
-        () -> {
-          final Optional<UserTask> userTask =
-              client
-                  .newUserTaskSearchRequest()
-                  .filter(userTaskSelector::applyFilter)
-                  .send()
-                  .join()
-                  .items()
-                  .stream()
-                  .filter(userTaskSelector::test)
-                  .findFirst();
+    LOGGER.debug(
+        "Mock: Complete user task [{}, userTaskKey: '{}'] with variables {}",
+        userTaskSelector.describe(),
+        userTask.getUserTaskKey(),
+        variables);
 
-          userTask.map(UserTask::getUserTaskKey).ifPresent(userTaskKey::set);
+    client.newCompleteUserTaskCommand(userTask.getUserTaskKey()).variables(variables).send().join();
+  }
 
-          assertThat(userTask)
-              .withFailMessage(
-                  "Expected to complete user task [%s] but no user task is available.",
-                  userTaskSelector.describe())
-              .isPresent();
-        });
+  @Override
+  public void completeUserTaskWithExampleData(final String elementId) {
+    completeUserTaskWithExampleData(UserTaskSelectors.byElementId(elementId));
+  }
 
-    LOGGER.debug(
-        "Complete user task with variables {} [user-task-key: '{}']", variables, userTaskKey.get());
-    client.newCompleteUserTaskCommand(userTaskKey.get()).variables(variables).send().join();
+  @Override
+  public void completeUserTaskWithExampleData(final UserTaskSelector userTaskSelector) {
+    final CamundaClient client = createClient();
+    final BpmnExampleDataReader exampleDataReader =
+        new BpmnExampleDataReader(client, awaitBehavior);
+
+    final UserTask userTask = awaitUserTask(userTaskSelector, client);
+    final String logPrefix =
+        String.format(
+            "Mock: Complete user task [%s, userTaskKey: '%s']",
+            userTaskSelector.describe(), userTask.getUserTaskKey());
+
+    try {
+      final String exampleData =
+          exampleDataReader.readExampleData(
+              userTask.getProcessDefinitionKey(), userTask.getElementId());
+
+      LOGGER.debug("{} with example data {}", logPrefix, exampleData);
+      client
+          .newCompleteUserTaskCommand(userTask.getUserTaskKey())
+          .variables(exampleData)
+          .send()
+          .join();
+    } catch (final BpmnExampleDataReaderException e) {
+
+      LOGGER.warn("{} without example data due to errors. {}", logPrefix, e.getMessage(), e);
```

Comment:
🔧 (same as above) Printing the error duplicated the failure message. We should remove the exception from the log message.


```suggestion
      LOGGER.warn("{} without example data due to errors. {}", logPrefix, e.getMessage());
```